Lib/test/datetimetester.py

```diff
@@ -1928,6 +1928,7 @@ def test_fromisoformat_fails(self):
             '2019-W53-1',       # No week 53 in 2019
             '2020-W54-1',       # No week 54
             '2009\ud80002\ud80028',     # Separators are surrogate codepoints
+            '0000W25',          # Invalid year
```

Comment:
Yes, it is optional, though I do think it would be better to use the form `'0000-W25-1'` , because it will be more obvious that it's failing because of the invalid value rather than because of some issue with the format (I think that's why I chose a clearly valid format with well-separated values for the tests in the first place).

I know it's a bit of a scope creep, but while we are here, it would probably be a good idea to round out the coverage to cover invalid day of week as well.

Here is a suggested patch that adds the extra test cases, and moves the surrogate test case to the end, so that all the "invalid component" tests are grouped together.

```suggestion
            '0000-W25-1',       # Invalid year
            '10000-W25-1',      # Invalid year
            '2020-W25-0',       # Invalid day-of-week
            '2020-W25-8',       # Invalid day-of-week
            '2009\ud80002\ud80028',     # Separators are surrogate codepoints
```

Reply:
I've updated the test cases as suggested.